lib/golem/system.h

```diff
@@ -0,0 +1,79 @@
+#ifndef GOLEM_SYSTEM_H
+#define GOLEM_SYSTEM_H
+
+#include <memory>
+#include <type_traits>
+#include <utility>
+
+namespace golem {
+
+// Models the hardware interface of your robot system
+// Accepts any class that implements the following functions for given Feedback and Input types:
+// - Feedback measure() const;
+// - void apply(const Input& input);
+template <typename Feedback, typename Input>
+class System {
+public:
+    System() = default;
+    System(const System&) = delete;
+    System& operator=(const System&) = delete;
+    System(System&&) = default;
+    System& operator=(System&&) = default;
+
+    template <typename T>
+    System(T&& impl)
+        : m_impl(new model_t<typename std::decay<T>::type>(std::forward<T>(impl)))
```

Comment:
meesa no like malloc calls()... Couldn't we just hold a reference to some external object ?

Reply:
I'm thinking to use CRTP instead of Sean Parent's runtime polymorphism. Trying to avoid inheritance at all costs :angel:

Reply:
- [ ] get rid of malloc

Reply:
Hmmm I think here inheritance is actually A-ok, isn't it ? And at least it will not blow in our face in strange ways ;)

Reply:
it's even less code with CRTP :heart_eyes_cat: